src/sql/src/plan/query.rs

```diff
@@ -2819,6 +2819,13 @@ fn plan_literal<'a>(l: &'a Value) -> Result<CoercibleScalarExpr, anyhow::Error>
             (Datum::Interval(i), ScalarType::Interval)
         }
         Value::String(s) => return Ok(CoercibleScalarExpr::LiteralString(s.clone())),
+        Value::Array(vals) => {
+            let coercible_vals = vals
+                .into_iter()
+                .map(|v| plan_literal(v))
+                .collect::<Result<Vec<CoercibleScalarExpr>, anyhow::Error>>()?;
+            return Ok(CoercibleScalarExpr::LiteralArray(coercible_vals));
```

Comment:
I think we should bail here to force folks to use the `ARRAY` syntax, if that makes sense!

Reply:
So like `WITH (start_offset = ARRAY[2,40000000])`? Or plan `WithOptions` separately?

Reply:
Like, I think if you just change this to `bail!("bare [] arrays are not supported in this context; use ARRAY[] or LIST[] instead")` then you'll be good to go. That means that

```
SELECT [1, 2, 3]
```

won't be a thing, but `WITH (start_offset = [1, 2, 3])` is ok. Does that make sense? Sorry, this is pretty hairy.

Reply:
It makes sense conceptually! Will figure this out, thanks!

Reply:
Yeah, so I think you need to plumb in a bool here that's like `is_array_allowed` and only set it to `true` if you're calling this function while planning a `WITH` option!